src/treeherder/job_handler.js

```diff
@@ -362,6 +362,11 @@ class Handler {
     });
   }
 
+  shouldReportExceptionRun(run) {
+    return (run.reasonCreated !== 'exception' &&
+            run.reasonResolved !== 'deadline-exceeded');
```

Comment:
I think the condition `run.reasonCreated !== 'exception'` should be enough... If a run is created to report an exception... then it's either cancel or deadline-exceeded.